devdoc/channel_internal.md

```diff
@@ -90,32 +109,36 @@ MOCKABLE_FUNCTION(, CHANNEL_RESULT, channel_internal_push, THANDLE(CHANNEL_INTER
 
 **SRS_CHANNEL_INTERNAL_43_099: [** `channel_internal_dispose` shall call `srw_lock_destroy`. **]**
 
+**SRS_CHANNEL_INTERNAL_43_165: [** `channel_internal_dispose` shall call `sm_destroy`. **]**
+
 ### channel_internal_pull
 ```c
-    MOCKABLE_FUNCTION(, CHANNEL_RESULT, channel_internal_pull, THANDLE(CHANNEL_INTERNAL), channel_internal, THANDLE(RC_STRING), correlation_id, PULL_CALLBACK, pull_callback, void*, pull_context, THANDLE(ASYNC_OP)*, out_op_pull);
+    MOCKABLE_FUNCTION(, CHANNEL_RESULT, channel_internal_pull, THANDLE(CHANNEL_INTERNAL), channel_internal, THANDLE(RC_STRING), correlation_id, ON_DATA_AVAILABLE_CB, on_data_available_cb, void*, pull_context, THANDLE(ASYNC_OP)*, out_op_pull);
 ```
 
-`channel_internal_pull` registers the given `pull_callback` to be called when there is data to be consumed.
+`channel_internal_pull` registers the given `on_data_available_cb` to be called when there is data to be consumed.
+
+**SRS_CHANNEL_INTERNAL_43_152: [** `channel_internal_pull` shall call `sm_exec_begin`. **]**
 
```

Comment:
I'm having serious misgivings about this style of sm use (calling `sm_exec_begin` in the sync function and `sm_exec_end` in the async callback.)